File: test/nodejs/duckdb_rdkit_test.js
var duckdb = require('../../duckdb/tools/nodejs');
var assert = require('assert');

describe(`duckdb_rdkit extension`, () => {
    let db;
    let conn;
    before((done) => {
        db = new duckdb.Database(':memory:', {"allow_unsigned_extensions":"true"});
        conn = new duckdb.Connection(db);
        conn.exec(`LOAD '${process.env.DUCKDB_RDKIT_EXTENSION_BINARY_PATH}';`, function (err) {
            if (err) throw err;
            done();
        });
    });

    it('duckdb_rdkit function should return expected string', function (done) {
        db.all("SELECT duckdb_rdkit('Sam') as value;", function (err, res) {
            if (err) throw err;
            assert.deepEqual(res, [{value: "DuckdbRdkit Sam 🐥"}]);
            done();
        });
    });

    it('duckdb_rdkit_openssl_version function should return expected string', function (done) {
        db.all("SELECT duckdb_rdkit_openssl_version('Michael') as value;", function (err, res) {
            if (err) throw err;
            assert(res[0].value.startsWith('DuckdbRdkit Michael, my linked OpenSSL version is OpenSSL'));
            done();
        });
    });
});

File: test/python/duckdb_rdkit_test.py
import duckdb
import os
import pytest

# Get a fresh connection to DuckDB with the duckdb_rdkit extension binary loaded
@pytest.fixture
def duckdb_conn():
    extension_binary = os.getenv('DUCKDB_RDKIT_EXTENSION_BINARY_PATH')
    if (extension_binary == ''):
        raise Exception('Please make sure the `DUCKDB_RDKIT_EXTENSION_BINARY_PATH` is set to run the python tests')
    conn = duckdb.connect('', config={'allow_unsigned_extensions': 'true'})
    conn.execute(f"load '{extension_binary}'")
    return conn

def test_duckdb_rdkit(duckdb_conn):
    duckdb_conn.execute("SELECT duckdb_rdkit('Sam') as value;");
    res = duckdb_conn.fetchall()
    assert res[0][0] == "DuckdbRdkit Sam 🐥"

def test_duckdb_rdkit_openssl_version_test(duckdb_conn):
    duckdb_conn.execute("SELECT duckdb_rdkit_openssl_version('Michael');");
    res = duckdb_conn.fetchall()
    assert res[0][0][0:51] == "DuckdbRdkit Michael, my linked OpenSSL version is OpenSSL"
